packages/xrpl/test/integration/transactions/nftokenMint.test.ts

```diff
@@ -1,67 +1,87 @@
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { testTransaction } from '../utils'
 
 // how long before each test case times out
 const TIMEOUT = 20000
 
 describe('NFTokenMint', function () {
-  // TODO: Once we update our integration tests to handle NFTs, replace this client with XrplIntegrationTestContext
+  let testContext: XrplIntegrationTestContext
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterEach(async () => teardownClient(testContext))
+
   it(
     'get NFTokenID',
     async function () {
-      const client = new Client('wss://s.altnet.rippletest.net:51233/')
-      await client.connect()
-
-      const { wallet, balance: _balance } = await client.fundWallet()
-
       const tx: NFTokenMint = {
         TransactionType: 'NFTokenMint',
-        Account: wallet.address,
+        Account: testContext.wallet.address,
         URI: convertStringToHex('https://www.google.com'),
         NFTokenTaxon: 0,
       }
-      try {
-        const response = await client.submitAndWait(tx, {
-          wallet,
-        })
-        assert.equal(response.type, 'response')
-        assert.equal(
-          (response.result.meta as TransactionMetadata).TransactionResult,
-          'tesSUCCESS',
-        )
+      const response = await testTransaction(
+        testContext.client,
+        tx,
+        testContext.wallet,
+      )
+      assert.equal(response.type, 'response')
 
-        const accountNFTs = await client.request({
-          command: 'account_nfts',
-          account: wallet.address,
-        })
+      const txRequest: TxRequest = {
+        command: 'tx',
+        transaction: hashSignedTx(response.result.tx_blob),
+      }
+      const txResponse = await testContext.client.request(txRequest)
 
-        const nftokenID =
-          getNFTokenID(response.result.meta as TransactionMetadata) ??
-          'undefined'
-        const accountHasNFT = accountNFTs.result.account_nfts.some(
-          (value) => value.NFTokenID === nftokenID,
-        )
+      assert.equal(
+        (txResponse.result.meta as TransactionMetadata).TransactionResult,
+        'tesSUCCESS',
+      )
 
-        assert.isTrue(
-          accountHasNFT,
-          `Expected to find an NFT with NFTokenID ${nftokenID} in account ${
-            wallet.address
-          } but did not find it.
-        \n\nHere's what was returned from 'account_nfts' for ${
-          wallet.address
-        }: ${JSON.stringify(accountNFTs)}`,
-        )
-      } finally {
-        await client.disconnect()
-      }
+      const accountNFTs = await testContext.client.request({
+        command: 'account_nfts',
+        account: testContext.wallet.address,
+      })
+
+      const nftokenID =
+        getNFTokenID(txResponse.result.meta as TransactionMetadata) ??
+        'undefined'
+
+      const accountHasNFT = accountNFTs.result.account_nfts.some(
+        (value) => value.NFTokenID === nftokenID,
+      )
```

Comment:
Since the account was just created for this test shouldn't `account_nfts` only have one NFT in it?

Reply:
I'm reusing the `testContext.wallet` so this account may have other nfts from other nft-related tests, so checking for `some` gives resilience against that race condition.

Reply:
`testContext.wallet` is regenerated for every test

Reply:
I see - I think using `.some` is still fine regardless - I'd rather not depend on the account only having one nft when the function returns an array.

Reply:
I'd still rather double check that, since if there's somehow another NFT in that account that'd mean something was wrong with our test framework, but I'm not going to hold up this PR because of that if you want to keep it.